backend/tracim_backend/models/tag.py

```diff
@@ -0,0 +1,60 @@
+from sqlalchemy import Column
+from sqlalchemy import ForeignKey
+from sqlalchemy import Sequence
+from sqlalchemy import Unicode
+from sqlalchemy import UniqueConstraint
+from sqlalchemy.orm import relationship
+from sqlalchemy.types import Integer
+
+from tracim_backend.models.data import Content
+from tracim_backend.models.data import User
+from tracim_backend.models.data import Workspace
+from tracim_backend.models.meta import DeclarativeBase
+from tracim_backend.models.mixins import CreationDateMixin
+
+
+class Tag(CreationDateMixin, DeclarativeBase):
+    __tablename__ = "tag"
+    __table_args__ = (UniqueConstraint("workspace_id", "tag_name"),)
+
+    tag_id = Column(Integer, Sequence("seq__tag__tag_id"), autoincrement=True, primary_key=True)
+    workspace_id = Column(
+        Integer,
+        ForeignKey("workspaces.workspace_id", onupdate="CASCADE", ondelete="CASCADE"),
+        nullable=False,
+    )
+    tag_name = Column(Unicode(), nullable=False)
+
+    workspace = relationship("Workspace", remote_side=[Workspace.workspace_id], lazy="joined")
+
+    author_id = Column(Integer, ForeignKey("users.user_id"), nullable=False)
+    author = relationship("User", remote_side=[User.user_id])
+
+    def __repr__(self):
+        return "<Tag(workspace_id=%s, tag_name=%s)>" % (
+            repr(self.workspace_id),
+            repr(self.tag_name),
+        )
+
+
+class TagOnContent(CreationDateMixin, DeclarativeBase):
+    __tablename__ = "content_tag"
+    __table_args__ = (UniqueConstraint("tag_id", "content_id"),)
+
+    id = Column(Integer, Sequence("seq__content_tag__id"), autoincrement=True, primary_key=True,)
```

Comment:
what is this id used for? according to the strategy which is understandable for this table (a many to many mapping table, I think this property is useless)

Reply:
This was making sqlalchemy fail a test because of the absence of a primary key, which I did not expect. I didn't really want to add this for the reasons you give. I'm open to suggestions.

Reply:
I understand the point. I see two ways to fix this:

- define a composite primary key (tag_id, content_id) - https://stackoverflow.com/questions/5835978/how-to-properly-create-composite-primary-keys-mysql
- if one of our supported sql engine does not support this, I suggest to comment the reason why you added the id column

Reply:
Your first point would be my preferred solution, and your second point, the second preferred. I'll look into it.

Reply:
It seems just marking both column as a primary_key=True in sqlalchemy is all we have to do to create a composite primary key. I hope it works on all the database engine.
https://learnings.desipenguin.com/post/composite-primary_key-in-sqlalchemy/